dagshub/upload/wrapper.py

```diff
@@ -0,0 +1,140 @@
+import requests
+import urllib
+import os
+from pprint import pprint
+from typing import Union
+from io import IOBase
+
+DEFAULT_SOURCE_URL = "https://dagshub.com/"
+CONTENT_UPLOAD_URL = "api/v1/repos/{owner}/{reponame}/content/{branch}/{path}"
+REPO_INFO_URL = "api/v1/repos/{owner}/{reponame}"
+
+def get_default_branch(src_url, owner, reponame):
+	res = requests.get(urllib.parse.urljoin(src_url, REPO_INFO_URL.format(
+		owner=owner,
+		reponame=reponame
+		)))
+	return res.json().get('default_branch')
+
+class Repo:
+	def __init__(self, owner, name, authToken=None, src_url=None, branch=None):
+		self.owner = owner
+		self.name = name
+		self.src_url = src_url if src_url is not None else os.environ.get("SRC_URL") if "SRC_URL" in os.environ else DEFAULT_SOURCE_URL
+
+		if authToken is not None:
+			self.authToken = authToken
+		elif "ACCESS_TOKEN" in os.environ:
+			self.authToken = os.environ["ACCESS_TOKEN"]
+		else:
+			raise Exception("Can't find access token. You can set an enviroment variable ACCESS_TOKEN with a DagsHub access token or pass it to Repo with 'authToken'.")
+		# TODO: verify token
+
+		if branch is not None:
+			self.branch = branch
+		else:
+			print("Branch wasn't provided. Fetching default branch...")
+			self._set_default_branch()
+		print("Set branch: ", self.branch)
+
+	def upload(self, file: Union[str, IOBase], message, versioning=None, new_branch=None, last_commit=None, path=None):
+		ds = DataSet(self, ".")
```

Comment:
It's not urgent now, but I think this is semantically wrong.
If I update some random dvc or git file, I'm not updating a dataset, I'm just uploading a file. DRY is good but the code still needs to tell a story that is correct.

Reply:
🤔